app/knight.py

```diff
@@ -0,0 +1,24 @@
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+
+    def using_armour(self, armour: list) -> int:
+        for part in armour:
```

Comment:
It's better to use a more specific type hint for armour. Instead of using list, use List[Dict[str, int]] from typing module to provide a more detailed type hint.

Reply:
If you use Python 3.10+ you can use list, dict in this case.